server/graphql/mutations.js

```diff
@@ -69,6 +76,26 @@ const mutations = {
         status: response.status,
         description: response.description
       }))
+
+      // record payment, if needed
+      .then(responseModel => {
+        if (response.user.card && response.user.card.token && tier.amount > 0) {
```

Comment:
if needed should be defined by amount, not by whether there is a user.card.
And we should return a proper error message if there isn't. Not clear by reading this code what are the possible error cases and messages